src/control/cmd/dmg/pretty/telemetry.go

```diff
@@ -0,0 +1,178 @@
+// and writes it to the io.Writer.
+func PrintMetricsQueryResp(resp *control.MetricsQueryResp, out io.Writer) error {
+	if resp == nil {
+		return errors.New("nil response")
+	}
+
+	if len(resp.MetricSets) == 0 {
+		return nil
+	}
+
+	for i, set := range resp.MetricSets {
+		if i > 0 { // a little space between metric sets
+			fmt.Fprintf(out, "\n")
+		}
+		fmt.Fprintf(out, "- Metric Set: %s (Type: %s)\n", set.Name, set.Type.String())
+
+		dw := txtfmt.NewIndentWriter(out)
+		fmt.Fprintf(dw, "%s\n", set.Description)
+
+		iw := txtfmt.NewIndentWriter(dw)
+		printMetrics(iw, set.Metrics, set.Type)
+
+	}
+	return nil
+}
+
+func printMetrics(out io.Writer, metrics []control.Metric, metricType control.MetricType) {
+	if len(metrics) == 0 {
+		fmt.Fprintf(out, "No metrics found\n")
+		return
+	}
+
+	nameTitle := "Metric"
+	labelTitle := "Labels"
+	valTitle := "Value"
+
+	tablePrint := txtfmt.NewTableFormatter(nameTitle, labelTitle, valTitle)
+	tablePrint.InitWriter(out)
+	table := []txtfmt.TableRow{}
+
+	for _, m := range metrics {
+		switch m.(type) {
+		case *control.SimpleMetric:
+			sm, _ := m.(*control.SimpleMetric)
+			labels := metricLabelsToStr(sm.Labels)
+			name := metricType.String()
+			table = append(table, txtfmt.TableRow{
+				nameTitle:  name,
+				labelTitle: labels,
+				valTitle:   fmt.Sprintf("%g", sm.Value),
+			})
+		case *control.SummaryMetric:
+			sm, _ := m.(*control.SummaryMetric)
+
+			labels := metricLabelsToStr(sm.Labels)
+			table = append(table, txtfmt.TableRow{
+				nameTitle:  "Sample Count",
+				labelTitle: labels,
+				valTitle:   fmt.Sprintf("%d", sm.SampleCount),
+			})
+			table = append(table, txtfmt.TableRow{
+				nameTitle:  "Sample Sum",
+				labelTitle: labels,
+				valTitle:   fmt.Sprintf("%g", sm.SampleSum),
+			})
+			for _, quant := range sm.Quantiles.Keys() {
+				table = append(table, txtfmt.TableRow{
+					nameTitle:  fmt.Sprintf("Quantile(%g)", quant),
+					labelTitle: labels,
+					valTitle:   fmt.Sprintf("%g", sm.Quantiles[quant]),
+				})
+			}
+		case *control.HistogramMetric:
+			hm, _ := m.(*control.HistogramMetric)
+
+			labels := metricLabelsToStr(hm.Labels)
+			table = append(table, txtfmt.TableRow{
+				nameTitle:  "Sample Count",
+				labelTitle: labels,
+				valTitle:   fmt.Sprintf("%d", hm.SampleCount),
+			})
+			table = append(table, txtfmt.TableRow{
+				nameTitle:  "Sample Sum",
+				labelTitle: labels,
+				valTitle:   fmt.Sprintf("%g", hm.SampleSum),
+			})
+			for i, bucket := range hm.Buckets {
+				name := fmt.Sprintf("Bucket(%d)", i)
+				table = append(table, txtfmt.TableRow{
+					nameTitle:  fmt.Sprintf("%s Upper Bound", name),
+					labelTitle: labels,
+					valTitle:   fmt.Sprintf("%g", bucket.UpperBound),
+				})
+				table = append(table, txtfmt.TableRow{
+					nameTitle:  fmt.Sprintf("%s Cumulative Count", name),
+					labelTitle: labels,
+					valTitle:   fmt.Sprintf("%d", bucket.CumulativeCount),
+				})
+			}
+		default:
```

Comment:
nit: this line doesn't do anything and can be removed

Reply:
Fixed. I was thinking of doing something for this default case, but couldn't figure out what that should be.